web-console/src/views/ingestion-view/ingestion-view.tsx

```diff
@@ -278,24 +282,30 @@ ORDER BY "rank" DESC, "created_time" DESC`;
     });
 
     this.taskQueryManager = new QueryManager({
-      processQuery: async capabilities => {
-        if (capabilities.hasSql()) {
-          return await queryDruidSql({
-            query: IngestionView.TASK_SQL,
+        processQuery: async capabilities => {
+          if (this.state.shouldOptimizeQuery) {
+            if (capabilities.hasOverlordAccess()) {
+              const resp = await Api.instance.get(`/druid/indexer/v1/tasks`);
+              return IngestionView.parseTasks(resp.data);
+            } else {
+              throw new Error(`must have SQL or overlord access`);
```

Comment:
why woudn't we want to fall back to the sql query if we can? I think that would make for a nicer experience.